common/isolationgroup/default-isolation-group-state.go

```diff
@@ -196,30 +292,65 @@ func isDrained(isolationGroup string, global types.IsolationGroupConfiguration,
 	return false
 }
 
-// A simple explicit deny-based isolation group implementation
-func availableIG(allIsolationGroups []string, global types.IsolationGroupConfiguration, domain types.IsolationGroupConfiguration) types.IsolationGroupConfiguration {
-	out := types.IsolationGroupConfiguration{}
-	for _, isolationGroup := range allIsolationGroups {
-		globalCfg, hasGlobalConfig := global[string(isolationGroup)]
-		domainCfg, hasDomainConfig := domain[string(isolationGroup)]
-		if hasGlobalConfig {
-			if globalCfg.State == types.IsolationGroupStateDrained {
+// ----- Mappers -----
+// dynamic config library does much of the decoding already, so this is asymmetric
+// to the serialization step. Incoming value should be just
+// map[string]interface{}
+func mapDynamicConfigResponse(in []*types.DynamicConfigEntry) (out types.IsolationGroupConfiguration, err error) {
+	if in == nil {
+		return nil, nil
+	}
+	out = make(types.IsolationGroupConfiguration)
+
+	for _, entry := range in {
+		for _, v := range entry.Values {
+			if v.Value == nil {
 				continue
 			}
-		}
-		if hasDomainConfig {
-			if domainCfg.State == types.IsolationGroupStateDrained {
-				continue
+			if v.Value.GetEncodingType() != types.EncodingTypeJSON {
+				return nil, fmt.Errorf("failed to decode values: %v, (%T)", v.Value, v.Value.EncodingType)
+			}
+			var partition types.IsolationGroupPartition
+			err := json.Unmarshal(v.Value.GetData(), &partition)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal entry: %v, got %w", string(v.Value.GetData()), err)
 			}
+			out[partition.Name] = partition
 		}
-		out[isolationGroup] = types.IsolationGroupPartition{
-			Name:  isolationGroup,
-			State: types.IsolationGroupStateHealthy,
+	}
+	return out, nil
+}
+
+func mapAllIsolationGroupsResponse(in []interface{}) ([]string, error) {
+	var allIsolationGroups []string
+	for k := range in {
+		v, ok := in[k].(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to get all-isolation-groups resonse from dynamic config: got %v (%T)", in[k], in[k])
 		}
+		allIsolationGroups = append(allIsolationGroups, v)
 	}
-	return out
+	return allIsolationGroups, nil
 }
 
-func fromCfgStore(in *persistence.FetchDynamicConfigResponse) (types.IsolationGroupConfiguration, error) {
-	panic("not implemented")
+func mapUpdateGlobalIsolationGroupsRequest(in types.IsolationGroupConfiguration) ([]*types.DynamicConfigValue, error) {
```

Comment:
I think it's not the right way to persist configuration. Why do we need to convert it to `types.DynamicConfigValue`?
Why not just store `types.IsolationGroupConfiguration`? If we do this, the data is more clean. We'll have something like 
```
{
  "schemaVersion": 0,
  "entries": [
    {
      "name": "",
      "values": [
        {
          "value": {
            "IsolationGroup": "zone-1",
            "State": 2
          }
        },
        {
          "value": {
            "IsolationGroup": "zone-2",
            "State": 1
          }
        }
      ]
    }
  ]
}
```